### PR TITLE
CVE-2014-1418 and CVE-2018-6188

### PR DESCRIPTION
**Pull request by dad9489**

### CODE REVIEW

File: cves/CVE-2014-1418.yml [hunk 1]
```diff
@@ -274,7 +309,15 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    This vulnerability was one that was obscure and difficult to notice, but
+    caused a pretty big potential issue. It went unnoticed for nearly 6 years,
```
Comment (sjc5897) [inline cves/CVE-2014-1418.yml]: "but caused a pretty big potential issue", this is a bit awkward phrasing. maybe try "but could cause a big issue"

File: cves/CVE-2014-1418.yml [hunk 2]
```diff
@@ -111,10 +117,14 @@ unit_tested:
...
     For the fix_answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  code:
-  code_answer:
-  fix:
-  fix_answer:
+  code: true
+  code_answer: |
+    The initial code involved adding functionality which was not initally tested.
+    In a later bug fix commit, unit tests for this code were added.
```
Comment (sjc5897) [inline cves/CVE-2014-1418.yml]: Did these later test find the vulnerability?

File: cves/CVE-2014-1418.yml [hunk 3]
```diff
@@ -274,7 +309,15 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
```
Comment (sjc5897) [inline cves/CVE-2014-1418.yml:313]: At what stage do you think this was this failure occurred or do you think it was just a Coding Mistake?

File: cves/CVE-2014-1418.yml [hunk 4]
```diff
@@ -287,8 +330,11 @@ CWE_instructions: |
 
   Just the number here is fine. No need for name or CWE prefix. If more than one
   apply here, then choose the best one and mention the others in CWE_note.
-CWE:
-CWE_note:
+CWE: 345
```
Comment (sjc5897) [inline cves/CVE-2014-1418.yml:337]: Upvote: 3

File: cves/CVE-2018-6188.yml [hunk 5]
```diff
@@ -285,7 +328,7 @@ CWE_instructions: |
 
   Just the number here is fine. No need for name or CWE prefix. If more than one
   apply here, then choose the best one and mention the others in CWE_note.
-CWE:
+CWE: 209
```
Comment (sjc5897) [inline cves/CVE-2018-6188.yml:331]: Upvote:2

File: cves/CVE-2014-1418.yml [hunk 6]
```diff
@@ -128,14 +138,17 @@ discovered:
...
+    James Westby. They noticed that cached data could be served do a different
+    session or to a user with no session at all.
+  automated: false
+  contest: false
+  developer: false
 autodiscoverable:
   instructions: |
     Is it plausible that a fully automated tool could have discovered
-    this? These are tools that require little knowledge of the domain,
+    this? These are tools that require little knowledge of tfhe domain,
```
Comment (jmc9961) [inline cves/CVE-2014-1418.yml]: 'tfhe' should be 'the'

File: cves/CVE-2014-1418.yml [hunk 7]
```diff
@@ -274,7 +309,15 @@ mistakes:
...
-  answer:
+  answer: |
+    This vulnerability was one that was obscure and difficult to notice, but
+    caused a pretty big potential issue. It went unnoticed for nearly 6 years,
+    but the conditions that allowed it to be exploited were rather rare especially
+    as time went on and less and less people used the older versions of IE that
+    caused the issue. If there were unit tests written when the code was first
+    added (instead of 3 years later), that might have helped. In addition, code
+    reviews could have helped to idenify the issue, as someone else looking at
+    the code could help idenity a potential vulnerability being introduced.
```
Comment (jmc9961) [inline cves/CVE-2014-1418.yml]: Do you think a developer could reasonably have realized that this vulnerability was possible just by looking at the code, without actually trying to test the feature on an older version of IE?

File: cves/CVE-2018-6188.yml [hunk 8]
```diff
@@ -144,8 +153,13 @@ autodiscoverable:
 
     The answer field should be boolean. In answer_note, please explain
     why you come to that conclusion.
-  answer_note:
-  answer:
+  answer_note: |
+    Yes, this could have been found by better unit and regression testing. An
+    issue from a regression is what caused this vulnerability to occur, so
+    doing regression testing could have revealed this issue. In addition,
+    having more robut unit tests in the first place that tested for inactive
```
Comment (jmc9961) [inline cves/CVE-2018-6188.yml]: 'robut' should be 'robust'

File: cves/CVE-2018-6188.yml [hunk 9]
```diff
@@ -177,20 +191,30 @@ subsystem:
     In the name field, a subsystem name (or an array of names)
 
     e.g. clipboard, model, view, controller, mod_dav, ui, authentication
-  answer:
-  name:
+  answer: |
+    This took place in the forms subsystem. The files with the relevant code were
+    named forms.py and test_forms.py. This code dealt with the authentication of
+    the form, specifically with logging in as a user. This all takes places within
```
Comment (jmc9961) [inline cves/CVE-2018-6188.yml]: 'places' should be 'place'

File: cves/CVE-2018-6188.yml [hunk 10]
```diff
@@ -255,8 +288,13 @@ lessons:
     applies:
     note:
   yagni:
-    applies:
-    note:
+    applies: true
+    note: |
+      This vulnerability was caused by a special case for an error message when
+      an account is marked as inactive. While this might be nice to have, it ended
+      up intoducing a vulnerability and the fix was just to remove the special
```
Comment (jmc9961) [inline cves/CVE-2018-6188.yml]: 'intoducing' should be 'introducing'

File: cves/CVE-2018-6188.yml [hunk 11]
```diff
@@ -272,7 +310,12 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    I believe that the main issue that created this vulnerability was a lack of
+    adequete regression testing. This piece of code did work as intended when it
```
Comment (jmc9961) [inline cves/CVE-2018-6188.yml]: 'adequete' should be 'adequate'

File: cves/CVE-2018-6188.yml [hunk 12]
```diff
@@ -285,7 +328,7 @@ CWE_instructions: |
 
   Just the number here is fine. No need for name or CWE prefix. If more than one
   apply here, then choose the best one and mention the others in CWE_note.
-CWE:
+CWE: 209
 CWE_note:
```
Comment (jmc9961) [inline cves/CVE-2018-6188.yml:332]: Upvote: 1

File: cves/CVE-2014-1418.yml [hunk 13]
```diff
@@ -287,8 +330,11 @@ CWE_instructions: |
 
   Just the number here is fine. No need for name or CWE prefix. If more than one
   apply here, then choose the best one and mention the others in CWE_note.
-CWE:
-CWE_note:
+CWE: 345
+CWE_note: |
```
Comment (jmc9961) [inline cves/CVE-2014-1418.yml:338]: Upvote: 1

**Pull request merged**